cyberpandas/ip_array.py

```diff
@@ -305,15 +305,23 @@ def isin(self, other):
             except ValueError:
                 networks.append(ipaddress.IPv6Network(net))
 
-        # TODO: perf
-        pyips = self.to_pyipaddress()
         mask = np.zeros(len(self), dtype='bool')
         for network in networks:
-            for i, ip in enumerate(pyips):
-                if ip in network:
-                    mask[i] = True
+            mask |= self._isin_network(network)
         return mask
 
+    def _isin_network(self, other):
+        # type: (Union[ipaddress.IPv4Network,ipaddress.IPv6Network]) -> ndarray
+        """Check whether an array of addresses is contained in a network."""
+        # A network is bounded below by 'network_address' and
+        # above by 'broadcast_address'.
+        # IPArray handles comparisons between arrays of addresses, and NumPy
+        # handles broadcasting.
+        net_lo = type(self)([other.network_address])
+        net_hi = type(self)([other.broadcast_address])
+
+        return (net_lo <= self) & (self <= net_hi)
```

Comment:
Is this using `ipaddress`'s internal comparisons, or is this implemented by us elsewhere? I like that it ended up as a simple one-liner.

Reply:
Implemented by us in https://github.com/ContinuumIO/cyberpandas/blob/52e28f763519d55e78f97f62cbb6d0e59c3efa9d/cyberpandas/ip_array.py#L185.

Amazing what a night of sleep can do to change your perspective on a problem :)